### PR TITLE
User Guide

### PR DESCRIPTION
**Pull request by adamschaub**
User Guide. Provide examples of real usage for Plugins/Services.

### CODE REVIEW

File: _docs/plugins-services.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ Input
 Output
   - List of typed/named outputs provided by the plugin
 
-Driver
-  - Provide multiple options for drivers (docker, lxd, exec-script/binary). Docker takes first priority.
+Image
+  - Container image url to run service operations
```
Comment (meganokeefe) [inline _docs/plugins-services.md]: no longer allowing exec services? containers only?

File: _docs/user-guide.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+Plugin - A template for Deploying, Statusing, Updating, and Destroying a single type of Service.
+
+Service - A collection of functionalities that can be used by clients. Each Service defines a set of Inputs and Outputs.
+
+Input - A required piece of information needed to operate on a Service.
+
+Output - A piece of information provided by an active Service.
+
+Dependency Graph - The implied relationship between services. Inferred from the relation of Inputs to Outputs of declared services
```
Comment (meganokeefe) [inline _docs/user-guide.md]: period @ end of sentence

File: _docs/user-guide.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+The Plugin foo/v1 specifies three required inputs: a list of "storage-urls", a target-ipv4, and a log-level. The log-level has been provided a default of "error". Two outputs will be provided: A list of "foo-urls", and a foo-observation. The container image `cisco/foo:v0.1` contains the functionality for Deploying/Statusing/Updating/Destroying a foo service.
+
+The Plugin bar/v1 specifies two required inputs: a list of "foo-urls", and a foo-observation. It provides no outputs. The container image `cisco/bar:v0.1` contains the functionality for Deploying/Statusing/Updating/Destroying a bar service.
+
+
+## Services
+
+Every Service uses a specific type/version of Plugin to allow it to Deploy/Status/Update/Destroy. Services provide a Plugin with the Inputs it requires, and can add additional items to a Plugins Outputs. 
+
+Each Service Input directly maps to one of its Plugin's Inputs. The value of a Service Input can be specified in two ways: statically assigned value, or inheret value from another service.
```
Comment (meganokeefe) [inline _docs/user-guide.md]: inheret->inherit

File: _docs/user-guide.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  type: list/foo-urls
+  from:
+    - foo.foo1.endpoints
+    - foo.foo2.endpoints
+- name: foo-observation
+  from: foo.foo1.another-observation
+image: sigma/bar:v0.1
+```
+
+The Service bar2 uses the bar Plugin, version1. It inherets input values from other Services by specifying `<plugin-name>.<service-name>.<output-name>`. For input foos, a list of foo-urls is required. Here two sources are specified. Their respective values will be concatenated into a single list. For input foo-observation, we use the value specified for another-observation.
```
Comment (meganokeefe) [inline _docs/user-guide.md]: inheret->inherit

Comment (meganokeefe) [inline _docs/user-guide.md]: code-ify `food-observation` and `another-observation`

File: _docs/plugins-services.md (hunk 1 above)
Comment (adamschaub) [inline _docs/plugins-services.md]: according to @mcaulfield

File: _docs/user-guide.md (hunk 2 above)
Comment (adamschaub) [inline _docs/user-guide.md]: done

File: _docs/user-guide.md (hunk 3 above)
Comment (adamschaub) [inline _docs/user-guide.md]: done

File: _docs/user-guide.md (hunk 4 above)
Comment (adamschaub) [inline _docs/user-guide.md]: done

Comment (adamschaub) [inline _docs/user-guide.md]: Done. Circled back to other variables and types as well.

**Pull request merged**